lib/req_ch.ex

```diff
@@ -1,12 +1,167 @@
+
+  import Req.Request, only: [get_option: 3]
+
   @doc """
   Attach this plugin to a Req Request.
+
+  ## Options
+
+    * `:scheme` - Required. Either `"http"` or `"https"`. Default is `"http"`.
+    * `:username` - Optional. Default is `nil`.
+    * `:password` - Optional. Default is `nil`.
+    * `:hostname` - Required. Default is `"localhost"`.
+    * `:port` - Required. Default is `"8123"`.
+    * `:clickhouse` - Required. The query to be performed. If not provided,
+      the request and response pipelines won't be modified.
+    * `:format` - Optional. The format of the response. Default is `:tsv`.
+      This option accepts `:tsv`, `:csv` or `:explorer` as atoms.
+
+      It also accepts all formats described in the #{@formats_page} page.
+
+      The `:explorer` format is special, and will build an Explorer dataframe
+      in case the `:explorer` dependency is installed.
+
+  ## Examples
+
+  With a plain query:
+
+      iex> req = Req.new() |> ReqCh.attach()
+      iex> Req.post!(req, clickhouse: "SELECT number from system.numbers LIMIT 3").body
+      "0\\n1\\n2\\n"
+
+  Changing the format to `:explorer` will return a dataframe:
+
+      iex> req = Req.new() |> ReqCh.attach()
+      iex> Req.post!(req, clickhouse: "SELECT number from system.numbers LIMIT 3", format: :explorer).body
+      #Explorer.DataFrame<
+        Polars[3 x 1]
+        number u64 [0, 1, 2]
+      >
+
   """
-  def attach(%Req.Request{} = request, _opts \\ []) do
+  def attach(%Req.Request{} = request, opts \\ []) do
+    request
+    |> Req.Request.prepend_request_steps(clickhouse_run: &run/1)
+    |> Req.Request.register_options(@connection_options ++ @general_options)
+    |> Req.Request.merge_options(opts)
+  end
+
+  defp run(%Req.Request{private: %{clickhouse_format: _}} = request), do: request
+
+  defp run(%Req.Request{options: %{clickhouse: _query}} = request) do
+    url_parts = [
+      get_option(request, :scheme, "http"),
+      "://",
+      maybe_credentials(request),
```

Comment:
instead of sending the credentials in the URL (which could show up in logs and stuff) I wonder if we can use the `auth: {:basic, "user:pass"}` feature.